backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/MetadataUtils.java

```diff
@@ -677,11 +679,37 @@ public static User loadUserMetadata(SqlDatabase db, String userName) {
     if (userRecord != null) {
       User result = new User(db, userName);
       result.setSettings(userRecord.get(SETTINGS, Map.class));
+      Map<String, String> roles = userRecord.get("roles", Map.class);
+      if (roles != null) {
+        result.setRoles(roles);
+      }
       return result;
     }
     return null;
   }
 
+  public static List<Member> loadUserRoles(SqlDatabase db) {
```

Comment:
recommend better place is SqlDatabase class? Then you can expose the function also in Database interface.